docs/frameworks/index.rst

```diff
@@ -0,0 +1,55 @@
+.. _spec:
+
+=======================
+Framework specification
+=======================
+
+The role of web-poet is to define a standard on how to write the extraction
+logic for a webpage, and allow it to be reused in different web scraping
+frameworks.
+
+:ref:`Page objects <page-objects>` should be flexible enough to be used with:
+
+* synchronous or asynchronous frameworks, callback-based and
+  ``async def / await`` based,
+* single-node and distributed systems,
+* different underlying HTTP implementations - or without HTTP support
+  at all, etc.
+
+To support web-poet in a web scraping framework, you need to:
+
+-   Support returning a :ref:`page object <page-objects>` based on a given
+    target URL and a desired :ref:`output item type <items>`.
+
+    To do that:
+
+    -   Use ``web_poet.default_registry`` to determine the right :ref:`page
+        object class <page-object-classes>` to use.
+
+        Optionally, let users indicate their own
+        :class:`~web_poet.rules.RulesRegistry` object.
+
+    -   Support building :ref:`input objects <inputs>` based on type hints,
```

Comment:
This deserves its own point. I think that's the most important thing a framework should implement. 

Some time ago scrapy-poet didn't support returning item classes directly; it also didn't support additional requests, and not all standard inputs are supported even now. I would still consider such framework a framework, as soon as it can build an instance of a page object, given a page object class. 

Everything else should probably be optional for a framework. 
It's also nice if a framework fails clearly when some standard features are not supported.

Reply:
https://github.com/scrapinghub/web-poet/pull/152/commits/3baff1225d325685e36c9d84467822da382c240b
https://github.com/scrapinghub/web-poet/pull/152/commits/0553b6c6e0745c29cfb35e5be319c604665d4cc7